install_files/securedrop-keyring/DEBIAN/preinst

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+# Solution adapted from DKG's work on `deb.torproject.org-keyring`.
+# In SecureDrop versions before 0.3.10, the Ansible playbooks used
+# the apt_key module to add the signing key to /etc/apt/trusted.gpg.
+# It's cleaner to use the trusted.gpg.d subdirectory, since we can
+# update that trivially in future versions of the keyring package.
+#
+# Therefore let's clean up prior versions of the key installed
+# to the general apt keyring, to ensure we only have one signing key
+# installed for authenticating securedrop-related packages.
+
+if [ -e /etc/apt/trusted.gpg ] && which gpg >/dev/null; then
+   (
+   h="$(mktemp -d)"
+   trap "rm -rf '$h'" EXIT
+
+   if gpg --homedir="$h" \
+          --batch --no-tty --no-default-keyring --keyring /etc/apt/trusted.gpg \
+          --list-key 0xB89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818 > /dev/null 2>&1 ; then
+      gpg --homedir="$h" \
+          --batch --no-tty --no-default-keyring --keyring /etc/apt/trusted.gpg \
+          --no-auto-check-trustdb \
+          --delete-key 0xB89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818 || true
```

Comment:
Continuing with my shell style comments: `readonly KEY_FPR='0xB89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818'` should be initialized right after `set -e`. If you want to use the keyid from an Ansible variable this should definitely be a Python script, which would bring us one step closer to resolving #1409 as well.